### PR TITLE
Make the whole package pass the checks

### LINKED ISSUE #7: Wireguard and Wireguard UI use the same port
I looked in the install script and it seems that Wireguard (VPN) and Wireguard UI (web server) use the same port, I wanted to know if it's a problem or not ? For me I think it is.

https://github.com/YunoHost-Apps/wireguard_ynh/blob/testing/scripts/install#L63

### PR DESCRIPTION
**Pull request by tytan652**
Closes #7 

## Seriously
- *I want to make this package really works*
- *Right now, only upgrade seems to not work, upgrade successful but get 502 bad gateway with package_check*
- *Others scripts may be incomplete*
- *Those change doesn't conflict with #5*

## Issue
- *The root installation test seems to not support private only package, so I needed to add is_public to the package, by default set to false in the manifest and set to true in check_process*
- *The package_linter ask to add User= to wireguard.service, and I'm not used to create systemd service*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested (manually).
- [x] Can be reviewed and tested.

## Package_check results

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/wireguard_ynh%20PR8%20(tituspijean)/)

### CODE REVIEW

**Comment by tytan652** [comment]
- *Update from version before adding is_public may not be testable*

**Comment by tituspijean** [comment]
We'll see that with the continuous integration checks. 🤞 
Thanks for your involvement!

**Comment by tytan652** [comment]
Your welcome

**Comment by tytan652** [comment]
I'm actually preparing a VM to investigate the upgrade bad gateway issue.

**Comment by tytan652** [comment]
I installed my branch of wireguard and update it with the same branch on my VM and I don't get a 502 bad gateway like the package_check.

**Comment by tytan652** [comment]
Maybe something is wrong in the check_process, I'm investigating this.

**Comment by tytan652** [comment]
>     * _Upgrade from version/commit before adding is_public may not be testable_

I'm talking about a upgrade check with a commit hash from before adding is_public, in the upgrade script I will need to add is_public and set it to 0 (private), but for the test I need to set it to 1 (public). I don't know if I can add a workarround for only this upgrade check.

**Comment by tituspijean** [comment]
> I installed my branch of wireguard and update it with the same branch on my VM and I don't get a 502 bad gateway like the package_check.

502 error may be due to the service not being fully started. Try checking for the service status before moving on (the helper has an argument for that), or add a `sleep X` to let it start up fully.

> I'm talking about a upgrade check with a commit hash from before adding is_public, in the upgrade script I will need to add is_public and set it to 0 (private), but for the test I need to set it to 1 (public). I don't know if I can create a workarround for only this upgrade check.

Since the app is in its early stages, and is currently private by default, I think we can skip this upgrade check. No need to add a whole `if` block only to make package check happy. If the user wants to make it public (as strongly discouraged it is), they can do it manually.

**Review by tituspijean — CHANGES_REQUESTED** [review]
*Disregard this one, I forgot how to* GitHub.

File: scripts/upgrade [hunk 1]
```diff
@@ -18,10 +18,18 @@ app=$YNH_APP_INSTANCE_NAME
...
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 
+# Change port for WireGuard UI because 8095 is default port for WireGuard VPN
+if [ $port -eq 8095 ]
+then
+	ynh_app_setting_delete --app=$app --key=port
+	port=$(ynh_find_port --port=8096)
+	ynh_app_setting_set --app=$app --key=port --value=$port
+fi
+
```
Comment (tituspijean) [inline scripts/upgrade:32]: It is only cosmetic, but can you put this code under a dedicated section?

```
#=================================================
# ENSURE DOWNWARD COMPATIBILITY
#=================================================
ynh_script_progression --message="Ensuring downward compatibility..." --time --weight=1
...
```

**Review by tituspijean — CHANGES_REQUESTED** [review]

File: scripts/upgrade (hunk 1 above)
Comment (tytan652) [reply scripts/upgrade:32]: I will do it in the next commit

**Comment by tytan652** [comment]
b0a0322 needs to be reviewed, I staged more changes than I wanted to by accident

**Comment by tytan652** [comment]
I found the issue for the 502 Bad Gateway, in the upgrade script I don't know why but WireGuard UI  doesn't care about the bind address and use the port 5000
```2020-12-30 13:20:47,171 DEBUG    yunohost.hook <lambda> - [877.1] + systemctl status wireguard_ui
2020-12-30 13:20:47,171 DEBUG    yunohost.hook <lambda> - [877.1] ● wireguard_ui.service - WireGuard UI
2020-12-30 13:20:47,172 DEBUG    yunohost.hook <lambda> - [877.1]    Loaded: loaded (/etc/systemd/system/wireguard_ui.service; enabled; vendor preset: enabled)
2020-12-30 13:20:47,172 DEBUG    yunohost.hook <lambda> - [877.1]    Active: active (running) since Wed 2020-12-30 13:19:47 UTC; 1min 0s ago
2020-12-30 13:20:47,172 DEBUG    yunohost.hook <lambda> - [877.1]  Main PID: 7086 (wireguard-ui)
2020-12-30 13:20:47,173 DEBUG    yunohost.hook <lambda> - [877.1]     Tasks: 6 (limit: 4696)
2020-12-30 13:20:47,173 DEBUG    yunohost.hook <lambda> - [877.1]    Memory: 12.9M
2020-12-30 13:20:47,173 DEBUG    yunohost.hook <lambda> - [877.1]    CGroup: /system.slice/wireguard_ui.service
2020-12-30 13:20:47,173 DEBUG    yunohost.hook <lambda> - [877.1]            └─7086 /opt/yunohost/wireguard/wireguard-ui --bind-address=127.0.0.1:8096 --disable-login >> /var/log/wireguard/ui.log 2>&1
2020-12-30 13:20:47,173 DEBUG    yunohost.hook <lambda> - [877.1] 
2020-12-30 13:20:47,174 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld systemd[1]: Started WireGuard UI.
2020-12-30 13:20:47,174 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: Wireguard UI
2020-12-30 13:20:47,174 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: App Version        : v0.2.5
2020-12-30 13:20:47,174 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: Git Commit        : c205a04443579a0e3de45526d772e76b4d3b441a
2020-12-30 13:20:47,175 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: Git Ref                : refs/tags/v0.2.5
2020-12-30 13:20:47,175 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: Build Time        : 12-30-2020 13:19:47
2020-12-30 13:20:47,175 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: Git Repo        : https://github.com/ngoduykhanh/wireguard-ui
2020-12-30 13:20:47,175 DEBUG    yunohost.hook <lambda> - [877.1] Dec 30 13:19:47 domain.tld wireguard-ui[7086]: ⇨ http server started on [::]:5000```

**Comment by tytan652** [comment]
It's also the wrong version

**Comment by tytan652** [comment]
Find the issue app.src exist and use 2.5 version

**Comment by tytan652** [comment]
We still need to tune the weight of the ynh_script_progression and get rid of the package_linter warning about wireguard.service

**Comment by tytan652** [comment]
Like I said I'm not really experienced with creation of systemd services, then I would like to know if I can add ```User=__APP__``` in wireguard.service

**Comment by tytan652** [comment]
Also shall we managed wg0.conf in backup/restore and remove

**Comment by tituspijean** [comment]
>Like I said I'm not really experienced with creation of systemd services, then I would like to know if I can add User=__APP__ in wireguard.service

This service calls `/bin/systemctl restart wg-quick@wg0.service`  which ASAIK can only be called as root. So I would not try it. ;)

> Also shall we managed wg0.conf in backup/restore and remove ?

I thought the UI took care of this, but maybe we can skip that and indeed directly handle backup and restoration of this file. It would make the package a bit more independent of the UI.

**Comment by tytan652** [comment]
Once I'm home, I add wg0.conf to the concerned scripts

**Comment by tytan652** [comment]
The file seems to be restored by the UI like you said, and it might be useless to restore it because WireGuard Ui will replaced it anyway.
The restored wg0.conf begins by that:
```
# This file was generated using wireguard-ui (https://github.com/ngoduykhanh/wireguard-ui)
# Please don't modify it manually, otherwise your change might got replaced.```

**Comment by tytan652** [comment]
After this PR is merged, I go back to DKMS changes

**Comment by tytan652** [comment]
Hello, is there any issue preventing the merge ?

**Comment by tituspijean** [comment]
None at all, sorry! I got distracted during family time. :)

**Comment by tytan652** [comment]
No problem

**Comment by tytan652** [comment]
@tituspijean, please could you bump YunoHost package version from 0.2.7~ynh1 to 0.2.7~ynh2, I think it's needed.

**Comment by tituspijean** [comment]
Sure ! But bear in mind that right now our changes are in testing branch, not master.

**Comment by tytan652** [comment]
I know but I will use this commit as a point of reference for upgrade check from a version with the backport kernel to the dkms one

**Pull request merged**